nextcord/guild.py

```diff
@@ -3463,3 +3467,204 @@ async def register_application_commands(self, *commands: BaseApplicationCommand)
+        enabled: bool = True,
+        exempt_roles: List[Role] = MISSING,
+        exempt_channels: List[GuildChannel] = MISSING,
+    ) -> AutoModerationRule:
+        ...
+
+    async def create_automod_rule(
+        self,
+        *,
+        name: str,
+        event_type: EventType = EventType.message_send,
+        trigger_type: TriggerType,
+        keyword_filters: List[str] = MISSING,
+        presets: List[KeywordPresetType] = MISSING,
+        notify_channel: GuildChannel = MISSING,
+        timeout_seconds: int = MISSING,
+        block_message: bool = True,
+        enabled: bool = True,
+        exempt_roles: List[Role] = MISSING,
+        exempt_channels: List[GuildChannel] = MISSING,
+    ) -> AutoModerationRule:
+        """|coro|
+
+        Create an auto moderation rule.
+
+        You must have the :attr:`~Permissions.manage_guild` to be able to do this.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The name of this auto moderation rule.
+        event_type: :class:`EventType`
+            The event type that this rule should start finding. Default to :attr:`EventType.message_send`.
+        trigger_type: :class:`TriggerType`.
+            The trigger type of this rule.
+        keyword_filters: List[:class:`str`]
+            The words that this rule should filter.
+        presets: List[:class:`KeywordPresetType`]
+            The preset keywords that this rule should filter.
+        notify_channel: :class:`abc.GuildChannel`
+            The channel that Discord will send an alert to this channel when the rule is triggered.
+            Either this or `timeout_seconds` must be provided.
+        timeout_seconds: :class:`int`
+            The seconds to timeout the user that triggered this rule.
+            Either this or `notify_channel` must be provided.
+        block_message: :class:`bool`
+            Whether or not the message that triggered the rule should be blocked when this rule is triggered. Default is True.
+        enabled: :class:`bool`
+            Whether or not this rule is enabled. Default is True.
+        exempt_roles: List[:class:`Role`]
+            The roles that should not be affected by this rule.
+        exempt_channels: List[:class:`abc.GuildChannel`]
+            The channels that should not be affected by this rule.
+
+        Returns
+        -------
+        :class:`AutoModerationRule`
+            The rule that was created.
+
+        Raises
+        -------
+        Forbidden
+            You do not have proper permissions to do this.
+        InvalidArgument
+            You specified both ``presets`` and ``keyword_filters``.
+        HTTPException
+            Creating the rule failed.
+        """
+        params = {
+            "name": name,
+            "event_type": event_type.value,
+            "trigger_type": trigger_type.value,
+            "enabled": enabled,
+        }
+        if keyword_filters is not MISSING and presets is not MISSING:
```

Comment:
This checks that not *both* are supplied. I think we also need a check to ensure at least *one* is provided no?